### ISSUE TITLE
Batched/full-batch transformer training under-converges (collapses at some batch sizes) vs per-example SGD

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1559]
## Summary
Transformer training on the small "canary" config converges with **per-example/SGD** updates (top-1 → 100%) but **plateaus or collapses** with batched / full-batch updates — and more epochs do **not** help. This affects every `AiModelBuilder.BuildAsync`-trained model, since the builder trains **batched** (one optimizer step per data-loader batch).

## Evidence (canary: vocab=8, batch=8, ctx=4, 8-example memorization set, seed=42)
Per-example (`DirectTrainAndMeasure`, 160 updates): **100%**.

Mini-batch sweep, 2000 epochs each (fresh seeded model per run):
| batch size | top-1 |
|---|---|
| 1 | 75% |
| 2 | **12.5% (= random chance — collapse to uniform output)** |
| 4 | 87.5% |
| 8 | 75% |

Full-batch (batch=8) top-1 is identical at 100 / 500 / 2000 / 5000 epochs (**75%**) — extra training does not help, so this is not undertraining.

## What is NOT the cause
- **Forward is correct.** A batched forward `[8,ctx]→[8,vocab]` matches per-row individual forwards to `7.4e-8` (float epsilon) — no cross-batch contamination.
- **Loss is standard.** `CategoricalCrossEntropyLoss.ComputeTapeLoss` sums over the class axis and averages over batch (PyTorch `reduction='mean'`), which is correct.
- Not licensing / weight streaming (sub-threshold no-op here).

## Likely area
Erratic, non-monotonic behavior across batch sizes plus a hard collapse to exactly random chance at batch=2 points at **training-dynamics / numerical stability** — candidate suspects: the Noam LR warmup schedule (warmup=10) interacting with small step counts, optimizer state, or representation/rank collapse — rather than a single localized gradient bug.

## Impact
Builder-trained models (batched) may under-train vs the documented per-example path. `ConfigureMethodTestBase` already warns: *"Per-example training is the known-convergent path. Don't use TrainBatched here."* — i.e. this was known but untracked.

## Repro
Subclass `ConfigureMethodTestBase`, train `MakeCanaryModel(seed:42)` on `MakeMemorizationSet()` with varying mini-batch sizes, measure `MeasureTrainingTopOne`.

**Comment by ooples** [comment]
## Root-caused + fixed (label smoothing) — branch `fix/1559-transformer-label-smoothing`

**Root cause.** The batched (averaged-gradient) path overshoots the Noam **warmup** LR into **hard softmax saturation**. Instrumenting the V=8 canary: batched reaches **100% top-1 around step 6**, then *regresses* and **freezes at 75%** — once the softmax pins to one-hot, its gradient vanishes (prediction spread pinned at `1.000000`) and the two wrong examples can never be corrected:

```
batched step  6: top1=100.0% spread=0.998
batched step  9: top1= 75.0% spread=1.000000   <- saturated, frozen
batched step 800: top1= 75.0% spread=1.000000   (identical at every step-count)
```

Per-example (SGD) avoids this — it advances the Noam step 8× faster and its gradient noise corrects the wrong examples before they saturate. Notably, the existing global-norm gradient clip (default on, norm 1.0) does **not** help: Adam normalizes the gradient (`step ≈ LR·m/√v`), so the overshoot is **LR-driven, not magnitude-driven**.

**Fix (paper-accurate).** "Attention Is All You Need" specifies **label smoothing ε=0.1**, which the library was missing. `CategoricalCrossEntropyLoss` now takes a `labelSmoothing` parameter (default `0.0` = unchanged) that replaces the one-hot target `y` with `(1-ε)·y + ε/K` across the tape / vector / GPU loss+gradient paths. Soft targets keep the optimal softmax off the 0/1 rail, so the gradient stays alive.

**Result (V=8 canary, batched):** frozen **75% → stable 87.5%**, softmax no longer saturated (spread `1.0 → ~0.88`). Realistic-scale batched (V=256) already converged and is unaffected. The residual (87.5% vs per-example 100%) is inherent batch-GD-vs-SGD dynamics on this pathologically tiny model.

**Delivered:** the loss change + a guard test (`Issue1559BatchedLabelSmoothingTests`) that trains the batched canary with/without smoothing and asserts smoothing strictly beats the frozen baseline, reaches ≥80%, and keeps the softmax off the hard rail. All existing `CategoricalCrossEntropy` tests stay green (default 0.0 is a no-op); both TFMs build.

**Open follow-up:** make label smoothing the Transformer's *default* (ε=0.1 per the paper) so `AiModelBuilder`-trained models get it automatically — deferred from this change because it shifts loss values for existing transformer tests and deserves its own scoped PR.
